File: app/src/main/java/c/m/storyapp/list_story/data/mapper/ListStoryMapper.kt
package c.m.storyapp.list_story.data.mapper

import c.m.storyapp.list_story.data.remote.dto.ListStoryResponseDTO
import c.m.storyapp.list_story.data.remote.dto.StoryDTO
import c.m.storyapp.list_story.domain.model.ListStoryResponse
import c.m.storyapp.list_story.domain.model.Story

fun ListStoryResponseDTO.toListStoryResponse(): ListStoryResponse {
    return ListStoryResponse(
        error = error,
        message = message,
        listStory = listStory?.map { it.toStory() }
    )
}

fun StoryDTO.toStory(): Story {
    return Story(
        id = id,
        name = name,
        description = description,
        photoUrl = photoUrl,
        createdAt = createdAt,
    )
}

File: lib/list_story/data/mapper/list_story_mapper.dart
import 'package:story_app/list_story/data/remote/dto/list_story_response_dto.dart';
import 'package:story_app/list_story/data/remote/dto/story_response_dto.dart';
import 'package:story_app/list_story/domain/model/list_story_response.dart';
import 'package:story_app/list_story/domain/model/story_response.dart';

extension StoryResponseDTOExtension on StoryResponseDTO {
  StoryResponse toStoryResponse() => StoryResponse(
        id: id,
        name: name,
        description: description,
        photoUrl: photoUrl,
        createAt: createAt,
      );
}

extension ListStoryResponseDTOExtension on ListStoryResponseDTO {
  ListStoryResponse toListStoryResponse() => ListStoryResponse(
        error: error,
        message: message,
        listStoryResponse: listStoryResponseDTO
            ?.map(
              (data) => data.toStoryResponse(),
            )
            .toList(),
      );
}
